src/dataflow-types/src/types.rs

```diff
@@ -1698,6 +1708,66 @@ pub mod sources {
             loader.load().await
         }
     }
+
+    struct SourceData(Result<Row, DataflowError>);
```

Comment:
I'm not really sure I see the point of this wrapper? I suppose it can make some code more readable, but I always prefer either `.0` or `.inner()` over `Deref` impls

Reply:
I can't implement `Codec` on `Result<Row, DataflowError>` due to orphan rules. I know `Deref` isn't the perfect fit here, I'm just trying to find the least painful way to make the orphan rule happy. I can remove the Deref impls until the time comes to actually encode data with it and see how it goes